tests/conftest.py

```diff
@@ -370,45 +371,24 @@ def set_oauth_credentials():
 
 
 @pytest.fixture
-def mock_post_collection(monkeypatch, set_oauth_credentials):
-    def _post_collection(th_client, project, oauth_key, oauth_secret,
-                         collection_inst, timeout=None):
-
-        jsondata = collection_inst.to_json()
-        signed_uri = th_client._get_project_uri(project,
-                                                collection_inst.endpoint_base,
-                                                data=jsondata, oauth_key=oauth_key,
-                                                oauth_secret=oauth_secret,
-                                                method='POST')
-        getattr(TestApp(application), 'post')(
-            signed_uri,
-            params=jsondata,
-            content_type='application/json'
-        )
+def mock_post_json(monkeypatch, set_oauth_credentials):
+    def _post_json(th_client, project, endpoint, json_data,
+                   timeout=None, auth=None):
 
-    from treeherder.client import TreeherderClient
-    monkeypatch.setattr(TreeherderClient, 'post_collection', _post_collection)
+        uri = th_client._get_project_uri(project, endpoint)
 
+        req = Request('POST', uri, data=json.dumps(json_data), auth=auth)
```

Comment:
I believe we can use the newish `json` argument, rather than needing `json.dumps()` with `data`:
http://docs.python-requests.org/en/latest/api/?highlight=json#lower-level-classes